File: backend/src/main/java/todoktodok/backend/notification/application/dto/response/NotificationItemResponse.java
package todoktodok.backend.notification.application.dto.response;

import java.time.LocalDateTime;
import todoktodok.backend.notification.domain.Notification;

public record NotificationItemResponse(
        NotificationData data,
        boolean isRead,
        LocalDateTime createdAt
) {

    public NotificationItemResponse(Notification notification) {
        this(
                new NotificationData(notification),
                notification.isRead(),
                notification.getCreatedAt()
        );
    }
}

File: android/TodokTodok/app/src/main/java/com/team/todoktodok/data/network/response/notification/NotificationItemResponse.kt
package com.team.todoktodok.data.network.response.notification

import com.team.domain.model.notification.Notification
import com.team.todoktodok.data.core.ext.toLocalDateTime
import kotlinx.serialization.Serializable

@Serializable
data class NotificationItemResponse(
    val data: NotificationContentResponse,
    val isRead: Boolean,
    val createdAt: String,
)

fun NotificationItemResponse.toDomain(): Notification =
    Notification(
        id = data.notificationId,
        notificationContent = data.toDomain(),
        isRead = isRead,
        createdAt = createdAt.toLocalDateTime(),
    )
